### PR TITLE
Added new unit property Display

### PR DESCRIPTION
**Pull request by ferinator**
We can use this as a draft and maybe display more stats from the unit in future (The design will probably also have to be adjusted).

### CODE REVIEW

File: game-frontend/src/UnitProperties.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    strokeWidth: 5,
+    circleOneStroke: '#FF0000',
+    circleTwoStroke: '#4C9900',
+  };
+
+  let unit = null;
+
+  if (units[0] !== undefined) {
+    unit = units[0] as UnitT;
+  }
```
Comment (StuckiSimon) [inline game-frontend/src/UnitProperties.tsx:26]: This currently uses the unit of the first user - but it should be the unit based on the `useUserContext()` `userId`.

File: game-frontend/src/UnitProperties.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  if (!gameState) {
+    return <p>Loading</p>;
+  }
```
Comment (StuckiSimon) [inline game-frontend/src/UnitProperties.tsx:30]: This will never get rendered.

File: game-frontend/src/UnitProperties.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  return (
+    <>{unit ? <UnitHealthStatus {...state} progress={unit.health} /> : null}</>
```
Comment (StuckiSimon) [inline game-frontend/src/UnitProperties.tsx:33]: No reason to wrap it in a React Fragment (`<>`). Simply do `return unit ? <UnitHealthStatus /> : null`

File: game-frontend/src/components/unitstatus/index.css [hunk 4]
```diff
@@ -0,0 +1,17 @@
+.circular-char {
```
Comment (StuckiSimon) [inline game-frontend/src/components/unitstatus/index.css:1]: We don't use separate css files in the project. Check `CodeBlock.tsx` for example.

File: game-frontend/src/components/unitstatus/index.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  useEffect(() => {
+    // progress -> what the user will give us (unit state in this case)
+    const progressOffset = ((100 - progress) / 100) * circumference;
+    setOffset(progressOffset);
+    // set the dependency to prevent infinte update (without offset, it always returns default value 0)
+  }, [setOffset, progress, circumference, offset]);
+
+  return (
+    <div>
+      <svg className="circular-char" width={size} height={size}>
```
Comment (sir-marc) [inline game-frontend/src/components/unitstatus/index.tsx:36]: Why do you work with svgs instead of the canvas?

**Review by sir-marc — CHANGES_REQUESTED** [review]

**Comment by sir-marc** [comment]
And in general, maybe a discussion we should have as a team:
I would prefer to display the health of every unit as a bar (or something) right on the unit itself so one can see how much health an enemy has.
@michaelschufi @StuckiSimon @ferinator @flipmess @mailaenderli

**Comment by sir-marc** [comment]
Additionally: The name of the PR should always beginn with your azure-ticket-id to connect the PR with the ticket directly. `AB#XX Your stuff`

File: game-frontend/src/components/unitstatus/index.tsx (hunk 5 above)
Comment (ferinator) [reply game-frontend/src/components/unitstatus/index.tsx:36]: I did some Tutorials before hand on how to create own progress bars and they used svg. I can try to replace it with canvas, but would need some help :)

Comment (sir-marc) [reply game-frontend/src/components/unitstatus/index.tsx:36]: Maybe try a tutorial working with canvas. And text us when you have questions.

**Comment by StuckiSimon** [comment]
As discussed in meeting this has been closed in favor of "healthbar per unit" visualization: https://github.com/code-and-conquer/darwin/pull/70

**Pull request closed without merging**